acme/challenge.go

```diff
@@ -427,6 +427,25 @@ func wireOIDC01Validate(ctx context.Context, ch *Challenge, db DB, jwk *jose.JSO
 	if err = db.UpdateChallenge(ctx, ch); err != nil {
 		return WrapErrorISE(err, "error updating challenge")
 	}
+
+	parsedIdToken, err := jwt.ParseSigned(wireChallengePayload.IdToken)
+	if err != nil {
+		return WrapErrorISE(err, "Invalid OIDC id token")
+	}
+	oidcToken := make(map[string]interface{})
+	if err := parsedIdToken.UnsafeClaimsWithoutVerification(&oidcToken); err != nil {
+		return WrapErrorISE(err, "Failed parsing OIDC id token")
+	}
+
+	orders, err := db.GetAllOrdersByAccountID(ctx, ch.AccountID)
+	if err != nil {
+		return WrapErrorISE(err, "Could not find current order by account id")
+	}
+
+	if err := db.CreateOidcToken(ctx, orders[0], oidcToken); err != nil {
```

Comment:
`orders[0]` looks like it could panic. You would have to safeguard this with a length check and an early exit if the length is zero. If not already done.